sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -296,11 +297,22 @@ case class UnresolvedFunction(
       copy(arguments = newChildren)
     }
   }
+
+  def name: FunctionIdentifier = multipartIdentifier.asFunctionIdentifier
```

Comment:
Where is this used? Since `asFunctionIdentifier` will throw exceptions for multi-part identifiers, it seems like this is dangerous. If name is used in normal places, those could suddenly start throwing errors.

Reply:
This is currently used in a few places in `Analyzer`. Let me see if I can replace these usages using the multi-part identifier.

Reply:
I removed this. Let me know if you have any other concern.